pkg/api/types.go

```diff
@@ -172,6 +172,9 @@ type Volume struct {
 	// This is optional for now. If not specified, the Volume is implied to be an EmptyDir.
 	// This implied behavior is deprecated and will be removed in a future version.
 	VolumeSource `json:",inline,omitempty"`
+	// SecretName is the name of a Secret in the same namespace as the pod containing this volume
+	// used for volume authentication
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
----- Original Message -----
> > @@ -172,6 +172,9 @@ type Volume struct {
> >  	// This is optional for now. If not specified, the Volume is implied to
> >  	be an EmptyDir.
> >  	// This implied behavior is deprecated and will be removed in a future
> >  	version.
> >  	VolumeSource `json:",inline,omitempty"`
> > +	// SecretName is the name of a Secret in the same namespace as the pod
> > containing this volume
> > +	// used for volume authentication
> > +	SecretName string `json:"secretName,omitempty"`
> 
> @smarterclayton can you provide an example of a volume that requires many
> secrets as opposed to just 1?

A GitVolumeSource that lists multiple repositories (in case one isn't available).  A URLVolumeSource that needs to hold a CACert (for TLS) and a username/password (basic auth).  Other things over HTTPS that need client certs *and* auth tokens.  A secret store that requires two forms of authentication (assertion + short lived token).

> 
> On one hand, I tend to think that nearly all volumes can potentially consume
> a secret and provide auth.  On the other hand, it feels a bit ham-fisted to
> make this applicable to all when our real use cases are exactly two (Ceph
> and SecretVolumeSource).
> 
> I think the real question is whether or not this applies generally for all
> volumes or if we just want to allow *some* volumes to have secrets.  The
> former is far less work (it's already achieved in this PR).  The latter
> would require an AuthenticatedVolumePlugin or some such interface construct,
> like we added with PersistentVolumePlugin.  The AuthVolumePlugin would have
> a SetSecret method.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7150/files#r29357972